app/poros/searched_user_poro.rb

```diff
@@ -0,0 +1,46 @@
+class SearchedUserPoro
+  attr_reader :id,
+              :first_name,
+              :last_name,
+              :is_remote,
+              :skills,
+              :distance
+
+  def initialize(searched_user, current_user)
+    @id = searched_user.id
+    @first_name = searched_user.first_name
+    @last_name = searched_user.last_name
+    @is_remote = searched_user.is_remote
+    @skills = build_skills(searched_user)
+    @distance = get_distance(searched_user, current_user)
+  end
+
+  private
+
+    def build_skills(user)
+      user.skills.map do |skill|
+        {
+          name: skill.name,
+          proficiency: skill.proficiency
+        }
+      end
+    end
+
+    def get_distance(searched_user, current_user)
+      r = 6371000
+      phi_1 = to_radian(searched_user[:lat])
+      phi_2 = to_radian(current_user[:lat])
+      d_phi = to_radian(current_user[:lat] - searched_user[:lat])
+      d_lam = to_radian(current_user[:lon] - searched_user[:lon])
+
```

Comment:
I'm not even going to try...But htis looks a lot like what I researched earlier.